webhooks/appmesh/gatewayroute_validator_test.go

```diff
@@ -1,14 +1,493 @@
+				Metadata: []appmesh.GRPCGatewayRouteMetadata{
+					{
+						Name: aws.String("service_name"),
+						Match: &appmesh.GRPCRouteMetadataMatchMethod{
+							Range: &appmesh.MatchRange{
+								Start: 4,
+								End:   8,
+							},
+						},
+					},
+				},
+			},
+			wantErr: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := validateMetadataIfAny(tt.currGR.Metadata)
+			if tt.wantErr != nil {
+				assert.EqualError(t, err, tt.wantErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_gatewayRouteValidator_validateHTTPRouteRewrite(t *testing.T) {
+	tests := []struct {
+		name    string
+		match   appmesh.HTTPGatewayRouteMatch
+		rewrite appmesh.HTTPGatewayRouteRewrite
+		wantErr error
+	}{
+		{
+			name: "Missing Rewrite filters",
+			match: appmesh.HTTPGatewayRouteMatch{
+				Prefix: aws.String("/"),
+			},
+			rewrite: appmesh.HTTPGatewayRouteRewrite{
+				Prefix:   nil,
+				Path:     nil,
+				Hostname: nil,
+			},
+			wantErr: errors.New("Either prefix, path or hostname for rewrite must be specified"),
+		},
+		{
+			name: "Both Prefix and Path Rewrite specified",
+			match: appmesh.HTTPGatewayRouteMatch{
+				Prefix: aws.String("/"),
+			},
+			rewrite: appmesh.HTTPGatewayRouteRewrite{
+				Prefix: &appmesh.GatewayRoutePrefixRewrite{
+					Prefix: aws.String("test-prefix"),
+				},
+				Path: &appmesh.GatewayRoutePathRewrite{
+					Exact: aws.String("test-prefix.com/payment"),
+				},
+			},
+			wantErr: errors.New("Both prefix and path cannot be specified. Only 1 allowed"),
+		},
+		{
+			name: "Incorrect Prefix Match format",
+			match: appmesh.HTTPGatewayRouteMatch{
+				Prefix: aws.String("/red"),
+			},
+			rewrite: appmesh.HTTPGatewayRouteRewrite{
+				Prefix: &appmesh.GatewayRoutePrefixRewrite{
+					Prefix: aws.String("/test/"),
+				},
+			},
+			wantErr: errors.New("Prefix to be matched on must end with '/'"),
+		},
+		{
+			name: "Incorrect Prefix Rewrite format",
+			match: appmesh.HTTPGatewayRouteMatch{
+				Prefix: aws.String("/red/"),
+			},
+			rewrite: appmesh.HTTPGatewayRouteRewrite{
+				Prefix: &appmesh.GatewayRoutePrefixRewrite{
+					Prefix: aws.String("/test"),
+				},
+			},
+			wantErr: errors.New("New Rewrite Prefix must end with '/'"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := validateHTTPRouteRewrite(&tt.rewrite, tt.match)
+			if tt.wantErr != nil {
+				assert.EqualError(t, err, tt.wantErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_gatewayRouteValidator_validateHostnameRewrite(t *testing.T) {
+	tests := []struct {
+		name    string
+		rewrite appmesh.GatewayRouteHostnameRewrite
+		wantErr error
+	}{
+		{
+			name: "Valid Hostname rewrite",
+			rewrite: appmesh.GatewayRouteHostnameRewrite{
+				DefaultTargetHostname: aws.String(string(enabled)),
+				TargetBased:           nil,
+			},
+			wantErr: nil,
+		},
+	}
```

Comment:
I don't see a test for `"Cannot specify both defaultTargetHostname and hostnameRewrite, only 1 is allowed"`

Reply:
Will update test files now and update the PR